kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/StrimziTestClient.java

```diff
@@ -45,13 +54,70 @@ public void produceMessages(String topicName, String bootstrap, String message,
+        String log;
+        try {
+            log = await().alias("Consumer waiting to receive messages")
+                    .ignoreException(KubernetesClientException.class)
+                    .atMost(timeout)
+                    .until(() -> {
+                        if (kubeClient().getClient().pods().inNamespace(namespace).withName(podName).get() != null) {
+                            return kubeClient().logsInSpecificNamespace(namespace, podName);
+                        }
+                        return null;
+                    }, m -> m.contains("Sending message:"));
+        }
+        catch (ConditionTimeoutException e) {
+            log = kubeClient().logsInSpecificNamespace(namespace, podName);
+            LOGGER.atInfo().setMessage("Timeout! Unable to produce the messages: {}").addArgument(log).log();
+        }
+        return log;
     }
 
     @Override
-    public String consumeMessages(String topicName, String bootstrap, String messageToCheck, int numOfMessages, Duration timeout) {
+    public List<ConsumerRecord<String, String>> consumeMessages(String topicName, String bootstrap, int numOfMessages, Duration timeout) {
         LOGGER.atInfo().log("Consuming messages using Strimzi Test Client");
         String name = Constants.KAFKA_CONSUMER_CLIENT_LABEL;
         Job testClientJob = TestClientsJobTemplates.defaultTestClientConsumerJob(name, bootstrap, topicName, numOfMessages).build();
-        return KafkaUtils.consumeMessages(topicName, name, deployNamespace, testClientJob, messageToCheck, numOfMessages, timeout);
+        String podName = KafkaUtils.createJob(deployNamespace, name, testClientJob);
+        String log = waitForConsumer(deployNamespace, podName, timeout);
+        LOGGER.atInfo().log(log);
+        List<String> logRecords = extractRecordLinesFromLog(log);
+        return getConsumerRecords(logRecords);
+    }
+
+    private String waitForConsumer(String namespace, String podName, Duration timeout) {
+        DeploymentUtils.waitForPodRunSucceeded(namespace, podName, timeout);
+        return kubeClient().logsInSpecificNamespace(namespace, podName);
+    }
+
+    private List<ConsumerRecord<String, String>> getConsumerRecords(List<String> logRecords) {
+        List<ConsumerRecord<String, String>> records = new ArrayList<>();
+        for (String logRecord : logRecords) {
+            ClientConsumerRecord clientConsumerRecord = ClientConsumerRecord.parseFromJsonString(logRecord);
+            if (clientConsumerRecord != null) {
+                ConsumerRecord<String, String> consumerRecord = clientConsumerRecord.toConsumerRecord();
+                records.add(consumerRecord);
+            }
+        }
+
+        return records;
+    }
+
+    private List<String> extractRecordLinesFromLog(String log) {
+        List<String> records = new ArrayList<>();
+        String stringToSeek = "Received message:";
```

Comment:
You want something like:

```java
Pattern.compile("^Received message:", Pattern.MULTILINE).matcher(testIllegal).replaceAll("")
```

the pattern object should be a constant (so the pattern is complied once).

The code is not quite perfect..  if the received message contains the string "\nReceived message:" the code would do the wrong thing.  For a better solution, I'd change the client not to emit the preamble.   I'm not suggesting you do that.. this is good enough!

Reply:
The problem with that solution is that we have a lot of content in the log that are not just the JSON that we want.
```
...
2024-06-07 06:07:03.933.2031041439114 INFO  SubscriptionState:396 - [Consumer clientId=my-consumer, groupId=my-group] Resetting offset for partition my-topic-03f15a-0 to position FetchPosition{offset=0, offsetEpoch=Optional.empty, currentLeader=LeaderAndEpoch{leader=Optional[my-cluster-kafka-2.my-cluster-kafka-brokers.kafka.svc:9092 (id: 2 rack: null)], epoch=0}}.
2024-06-07 06:07:04.105.2031213996425 INFO  KafkaConsumerClient:108 - Received message: {"timestamp":1717740414167,"timestampType":"CreateTime","topic":"my-topic-03f15a","partition":0,"offset":0,"key":null,"payload":"\u0000i\u000Fmy-topic-03f15avault:v1:koLuTR8g8dA0j5IcoqmLnUiDg3yvTDvvsnM5WojafM0xlYgDSrwxkKTfmv7giJWHg+iNC5RQYSZMj+um\u0000b??Up?(F?(\u000E????WV??Wk????\u0014!D\u0014T??W\"\u0000???m?@","headers":[{"kroxylicious.io/encryption":"\u0002"}]}
2024-06-07 06:07:05.152.2032260899289 INFO  KafkaConsumerClient:87 - Shutting down the executor
2024-06-07 06:07:05.153.2032261571786 INFO  KafkaConsumerClient:78 - All messages successfully received
```
and this was the simplest way I found to do that.

We have to think about that we don't always have clean logs where to read the JSON and that's all. I needed to go line by line to discard those lines which are not interesting for us.

Reply:
Okay, I didn't appreciate that you were having to scrape the output from the Strimzi Test Client's log4j log file.  Yuck.  It is odd you, wouldn't normally mix a commands output with its diagnostic log, but that's what we have. Ho hum..

Reply:
Yeah, getting info from logs is not usually easy, and here we have three different ways of logging with each kafka client... I did my best

Reply:
Yeah, I appreciate that.  Really I think Strimzi Test Client ought to just write what it consumes to stdout.  Logging is orthogonal concern.  We don't need to think about that change now.